src/gui/FxMixerView.cpp

```diff
@@ -598,29 +598,19 @@ void FxMixerView::updateFaders()
 
 	for( int i = 0; i < m_fxChannelViews.size(); ++i )
 	{
-		const float opl = m_fxChannelViews[i]->m_fader->getPeak_L();
-		const float opr = m_fxChannelViews[i]->m_fader->getPeak_R();
 		const float fallOff = 1.25;
-		if( m->effectChannel(i)->m_peakLeft >= opl/fallOff )
-		{
-			m_fxChannelViews[i]->m_fader->setPeak_L( m->effectChannel(i)->m_peakLeft );
-			// Set to -1 so later we'll know if this value has been refreshed yet.
-			m->effectChannel(i)->m_peakLeft = -1;
-		}
-		else if( m->effectChannel(i)->m_peakLeft != -1 )
-		{
-			m_fxChannelViews[i]->m_fader->setPeak_L( opl/fallOff );
-		}
-
-		if( m->effectChannel(i)->m_peakRight >= opr/fallOff )
-		{
-			m_fxChannelViews[i]->m_fader->setPeak_R( m->effectChannel(i)->m_peakRight );
-			// Set to -1 so later we'll know if this value has been refreshed yet.
-			m->effectChannel(i)->m_peakRight = -1;
-		}
-		else if( m->effectChannel(i)->m_peakRight != -1 )
-		{
-			m_fxChannelViews[i]->m_fader->setPeak_R( opr/fallOff );
-		}
+		
+		m_fxChannelViews[i]->m_fader->setPeak_L(
+			qMax( m->effectChannel(i)->m_peakLeft,
+			      m_fxChannelViews[i]->m_fader->getPeak_L()/fallOff)
```

Comment:
I am aware that it was not really introduced by this PR but IMO this is the wrong place for this logic. Why should the whole FX mixer view decide how the Faders behave with regards to the falloff?

The functionality should be moved into the `Fader`  class. The `Fader` should only be fed with the raw peak values and it's then up to the presentation code of the fader to decide how the display is rendered and updated, e.g. how much lag there is.

Reply:
- In principle, you're right.
- In practice, would I ever want a mixer with VU meters not having identical falloff? Don't think so. 
- Decentralizing the falloff would mean the Fader class probably needs some concept of time - a Qt signal or whatever way the FxMixerView gets its updates. Or then some contract of the _"don't refresh this more than once for every refresh period"_ kind, which I have a bad feeling about. Something will violate that sooner or later.
- I like making my PRs minimal, maybe too minimal sometimes. Possibly hoping it would speed up the process or something.

Feel free to put in your own PR and supersede mine or whatever, I don't mind.

Reply:
The problem is that with the current code it's not even possible to have meters with different falloffs. If someone wanted to implement an analog meter with a different falloff behavior then it would not be possible.

With regards to the concept of time: I wonder if it would not be better in the first place if the faders/meters would get the full contents of the sample buffers and then process them as they see fit, i.e. not even some pre-computation should be done outside. They should also know the current sample rate so that they could use it to compute the falloff factors or coefficients of low pass filters so that the falloff looks the same for all sample rates. Looking at the code it seems that currently they might behave and look differently for different sample rates, e.g. much more fidgety for high sample rates like 192 kHz.

I think minimal PRs are good because they are much better to review. :+1: